src/net_processing.cpp

```diff
@@ -2728,17 +2730,14 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
             // nodes)
             m_connman.PushMessage(&pfrom, msgMaker.Make(NetMsgType::SENDHEADERS));
         }
-        if (pfrom.GetCommonVersion() >= SHORT_IDS_BLOCKS_VERSION) {
-            // Tell our peer we are willing to provide version 1 or 2 cmpctblocks
+        if (pfrom.GetCommonVersion() >= SHORT_IDS_BLOCKS_VERSION &&
+            WITH_LOCK(cs_main, {return State(pfrom.GetId())->fHaveWitness;})) {
```

Comment:
Also, is this a behavioral change? It's not clear to me why the `fHaveWitness` check was added here.
Edit: Rather, why _wasn't_ it there before?

Reply:
I've removed the new check since it's not necessary.